Fix playlist card layout info (Closes #156)

pashz7 opened the pull request:
1. Changed props in component PlaylistCard in PlaylistsPage , MainPage 
2. Delete description 
3. Added a function that formats the AddedAt property according to the Figma design
4. Moved types of PlaylistCard to a separate file PlayListCardTypes  
5. add a button that navigate to the playlist ownerPage  

Property tracksCount info is still missing in the PlaylistAttributes, i have an idea how to solve this problem , but i think it will be better to fix it on backend
Now element Card is a Link maybe it is will be better to restrict the clickable area to the playlist image only to choose the current Playlist.  Maybe i am wrong
It would be great to discuss the PlaylistCard UI on the user’s page.

pashz7 committed:
Fix playlist card layout info (Closes #156)

Svyat-1418 commented on apps/rtk-query/src/features/playlists/ui/PlaylistCard/PlaylistCard.tsx:
```diff
@@ -8,41 +8,37 @@ import {
+import {formatCreatedDate} from "@/shared/utils/formatCreatedDate.ts";
+import type {PlaylistCardProps} from "@/shared/types";
 
-type PlaylistCardPropsBase = {
```
If you want to move the component props related types from the component .tsx file, it would be better to put them to a saparate file **_next to the componenent file_** (in this case I would do it in `playlist-card.types.ts`).

It's better to leave these types here, since them are used only for PlaylistCard component

Svyat-1418 commented on apps/rtk-query/src/pages/UserPage/ui/UserTabs/MyLikedPlaylistsTab/MyLikedPlaylistsTab.tsx:
```diff
@@ -54,7 +54,9 @@ export const MyLikedPlaylistsTab = () => {
                 title={playlist.attributes.title}
                 imageSrc={image?.url}
-                description={playlist.attributes.description}
+                userName={playlist.attributes.user.name}
```
```
    userName={playlist.attributes.user.name}
    userId={playlist.attributes.user.id}
    addedAt={playlist.attributes.addedAt}
```

<img width="1001" height="551" alt="Screenshot_396" src="https://github.com/user-attachments/assets/8de09bf5-7c87-4619-a7bd-6f3fec9488a1" />

Figma design shows that it's correct to display this info on **My Liked Playlists** tab, so you can leave it as it is.
However I supose that it's strange behavior

Svyat-1418 commented on apps/rtk-query/src/pages/UserPage/ui/UserTabs/PlaylistsTab/PlaylistsTab.tsx:
```diff
@@ -49,7 +49,9 @@ export const PlaylistsTab = () => {
                 title={playlist.attributes.title}
                 imageSrc={image?.url}
-                description={playlist.attributes.description}
+                userName={playlist.attributes.user.name}
```
<img width="1052" height="578" alt="Screenshot_397" src="https://github.com/user-attachments/assets/41e50417-5ff7-4690-8c93-3251e4f6497b" />

`userName`, `createdAt`, `<ReactionButtons />` are **_excessive_** if it's **My Playlists** tab according to the design

Svyat-1418 commented on apps/rtk-query/src/shared/types/playlistCardTypes.ts:
```diff
@@ -0,0 +1,23 @@
+import {CurrentUserReaction} from "@/shared/components";
```
I have already leave the review comment about PlaylistCard props types that you place here

Svyat-1418 commented on apps/rtk-query/src/shared/utils/formatCreatedDate.ts:
```diff
@@ -0,0 +1,14 @@
+        return 'today'
+    }
+
+    return `${differDays} ${differDays === 1 ? 'day' : 'days'} ago`
```
You should move static text to the translation instead of hardcode.

There's also text overflowing 

<img width="1369" height="479" alt="Screenshot_394" src="https://github.com/user-attachments/assets/b1e103b0-3d23-4407-a863-2d890866153b" />

It should be handled, for example with `text-overflow: ellipses;` or infinite line animation on hover or more... button whose behavior also can be different

It would be better to display created date in DD/MM/YYYY format or with powerfull and easyweighting library? for example date-fns

Svyat-1418 commented:
@pashz7 
As far as number of tracks is concerned, it is already task for backend.

You said that have some proposals about this

pashz7 reviewed (commented):
123

KirillDolzhenkov commented on apps/rtk-query/src/features/playlists/ui/PlaylistCard/PlaylistCard.tsx:
```diff
@@ -58,9 +54,25 @@ export const PlaylistCard = ({
-      <Typography variant="body3" className={s.description}>
-        {description}
-      </Typography>
+
```
Why did you remove the props description? I'm a little out of context, but is this component not reused anywhere else? Does it look the same everywhere? It looks like it could be passed from the outside as a slot. Maybe I'm wrong.

pashz7 committed:
Fixed display information and addetDate (Closes #156)

pashz7 committed:
Merge branch 'develop' into 156-display-the-information-corresponding-to-the-design-layout-on-the-playlist-card

pashz7 commented:
@Svyat-1418 
Fixed UI on UserPage/Playlists according Figma 
Added navigation to the playlist page when the image is clicked.
Added  properties to JSON object for translating the playlist creation date.
Fixed function that displays the playlist creation date depending on its creation date.

Svyat-1418 reviewed (changes requested):
@pashz7 
Rename your files with util functions from `camelCase` approach to `kebab-case`approach according to the **Naming** section from `README.md`.

Before perfoming this, update your local `develop` and then merge `develop` into your branch

Svyat-1418 commented on apps/rtk-query/src/features/playlists/ui/PlaylistCard/PlaylistCard.tsx:
```diff
@@ -1,77 +1,127 @@
+                                 userId,
+                                 addedAt,
+                                 isShowCurrentUser = true,
+                                 isShowCreatedDate = true,
```
- [x] - `isShowCurrentUser ` should be renamed to `shouldShowOwnerName`
- [x] - `isShowCreatedDate ` should be renamed to `shouldShowCreatedDate` 
- [x] - it's better to set default values of the renamed props above to `false`
- [x] - where owner name and/or Created date need to be displayed just pass to PlayllistCard only these props, for example `<PlayllistCard shouldShowCreatedDate  />`. It's the same as `<PlayllistCard shouldShowCreatedDate="true"  />`.

Svyat-1418 commented on apps/rtk-query/src/features/playlists/ui/PlaylistCard/PlaylistCard.tsx:
```diff
@@ -1,77 +1,127 @@
+                to={`/playlists/${id}`}
+                className={s.imageLink}
+                aria-label={`Open playlist ${title}`}
+            >
```
- [x] - `Paths` constant object shold be used here instead of hardcode.
- [x] - `aria-label` value needs to be moved to the translations

pashz7 committed:
Merge branch 'refs/heads/develop' into 156-display-the-information-corresponding-to-the-design-layout-on-the-playlist-card

# Conflicts:
#	apps/rtk-query/src/shared/translations/languages/en.json
#	apps/rtk-query/src/shared/translations/languages/ru.json

pashz7 committed:
fix utils function according to kebab-case

Svyat-1418 commented on apps/rtk-query/src/features/playlists/ui/PlaylistCard/PlaylistCard.tsx:
```diff
@@ -1,77 +1,127 @@
+                        <Link
+                            to={`/users/${userId}`}
+                            className={s.userLink}
+                            onClick={handleUserNameClick}
```
`to={`/users/${userId}`}` is excessive here: `handleUserNameClick` performs programming navigation

pashz7 commented:
@Svyat-1418 
fix naming of utils function according to kebab-case
and add reaction buttons in Tab LIkedPlaylists according to Figma

Svyat-1418 commented on apps/rtk-query/src/features/playlists/ui/PlaylistCard/PlaylistCard.tsx:
```diff
@@ -1,77 +1,127 @@
+            <div className={s.details}>
+                {isShowCurrentUser &&
+                    <div className={s.madeFor}>
+                        <Typography variant="body2" as="span">Made for </Typography>
```
add this text to the translations

Svyat-1418 commented on apps/rtk-query/src/features/playlists/ui/PlaylistCard/PlaylistCard.tsx:
```diff
@@ -1,77 +1,127 @@
+
+                <div className={s.detailsRow}>
+                    <Typography variant="body2" className={s.tracks}>
+                        0 tracks
```
`tracksCount` will be returned from the backend later, but `tracks` text needs to be taken from the translations.

Leave `TODO: ` for the `tracksCount`

pashz7 committed:
fix: translations,naming of props

pashz7 commented:
@Svyat-1418 
The suggested changes have been implemented. Please review when you have a moment.

Svyat-1418 commented on apps/rtk-query/src/shared/translations/languages/ru.json:
```diff
@@ -134,5 +137,13 @@
+    "tracks_count_one": "{{count}} трек",
+    "tracks_count_two": "{{count}} трека",
+    "tracks_count_few": "{{count}} трека",
+    "tracks_count_many": "{{count}} треков",
```
@pashz7  `tracks_count_two`, `tracks_count_many` may be excessive in this case.
Clarify me please your decision. It's quite possible that I'm mistaken.
Also you missed `made_for` for russian locale

pashz7 replied:
I think you're right. I just included all possible cases. I agree that tracks_count_two is a redundant property. I added tracks_count_many for the correct translation into Russian (e.g., 1035 tracks). And I made the other property in case whole number values are needed for some statistics or something. And I created the other property just in case fractional values are needed for some kind of statistics (for example, something like an average number of tracks).

Svyat-1418 commented on apps/rtk-query/src/shared/translations/languages/en.json:
```diff
@@ -131,5 +134,10 @@
     "created": "Created"
+  },
+  "playlist": {
+    "made_for": "Create for",
```
According to the Figma design, this should be `Made for` text

Svyat-1418 commented on apps/rtk-query/src/shared/configs/paths.ts:
```diff
@@ -2,6 +2,7 @@ export const Paths = {
   Main: '/',
   Auth: '/sign-in',
   Playlists: '/playlists',
+  Playlist: '/playlists',
```
I consider that the field `Playlist` is excessive because its value matches with the `Playlists` field value. It's better to use `to={`${Paths.Playlist}/${id}`}` in the `PlaylistCard` component

Svyat-1418 commented on apps/rtk-query/src/features/playlists/ui/PlaylistCard/PlaylistCard.tsx:
```diff
@@ -21,8 +19,8 @@ type PlaylistCardPropsBase = {
-    isShowCurrentUser?: boolean
-    isShowCreatedDate?: boolean
+    shouldShowOwnerName?: boolean
+    shouldShowCreateDate?: boolean
```
Here is a misspell: `shouldShowCreatedate` instead of `shouldShowCreateDate`. 

https://github.com/it-incubator/musicfun-react-all-stacks/pull/161#discussion_r2619622782

Svyat-1418 reviewed (changes requested):
Setting your IDE for applying `eslint` and `prettier` preferances for the project. 
Some local setting of yours sets large gaps in some files

<img width="1622" height="913" alt="Image" src="https://github.com/user-attachments/assets/6bbd5e83-4d58-4a56-987b-3d42426f2a7e" />

pashz7 committed:
fix: added missing properties to translation objects, renaming

Svyat-1418 reviewed (approved):
Ok, It's acceptable
